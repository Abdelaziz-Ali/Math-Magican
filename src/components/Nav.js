import React, { Outlet } from 'react';

function Nav() {
  return (
    <>
      <a href="/">
        <h1>Math magicians</h1>
      </a>
      <nav>
        <ul>
          <li>
            <a href="/">Home</a>
          </li>
          <li>
            <a href="/calculator">Calculator</a>
          </li>
          <li>
            <a href="/quote">Quote</a>
          </li>
        </ul>
      </nav>
      <Outlet />
    </>
  );
}
export default Nav;
