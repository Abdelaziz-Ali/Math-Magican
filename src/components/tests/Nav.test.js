import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Nav from '../Nav';

describe('Tests Nav component', () => {
  it('renders correctly', () => {
    const quotesPage = render(<BrowserRouter><Nav /></BrowserRouter>);
    expect(quotesPage).toMatchSnapshot();
  });
});
