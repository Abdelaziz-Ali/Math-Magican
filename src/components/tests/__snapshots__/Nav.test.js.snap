// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Nav component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav>
        <h1>
          <a
            href="/"
          >
            Math Magicians
          </a>
        </h1>
        <ul>
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
              style="color: rgb(245, 145, 62);"
            >
              Home
            </a>
          </li>
          <li>
            <span>
                |  
            </span>
          </li>
          <li>
            <a
              class=""
              href="/calculator"
            >
              Calculator
            </a>
          </li>
          <li>
            <span>
                |  
            </span>
          </li>
          <li>
            <a
              class=""
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav>
      <h1>
        <a
          href="/"
        >
          Math Magicians
        </a>
      </h1>
      <ul>
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            style="color: rgb(245, 145, 62);"
          >
            Home
          </a>
        </li>
        <li>
          <span>
              |  
          </span>
        </li>
        <li>
          <a
            class=""
            href="/calculator"
          >
            Calculator
          </a>
        </li>
        <li>
          <span>
              |  
          </span>
        </li>
        <li>
          <a
            class=""
            href="/quote"
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
